### ISSUE TITLE
2.3.0: syntax error in file lib/diplomat/rest_client.rb:243 ?

### ISSUE THREAD

**Issue by guilvey** [WeAreFarmGeek/diplomat#204]
Hello,
Since the release of diplomat 2.3.0, I am having a ruby issue in file rest_client.rb:243:

```
/usr/local/share/gems/gems/diplomat-2.3.0/lib/diplomat/rest_client.rb:261: syntax error, unexpected '.'
...       rest_options[:headers]&.map { |k, v| req.headers[k.to...
...                               ^
```

Can you have a look ?
Thank you,
Guillaume.

**Comment by pierresouchay** [comment]
@guilvey which version of Ruby are you running?

**Comment by guilvey** [comment]
Hello @pierresouchay ,
Thanks for your quick reply, I am using ruby 2.0.0p648 (2015-12-16) [x86_64-linux]
Guillaume.

**Comment by pierresouchay** [comment]
Hello @guilvey,

In 2.3.0, I did fix support for Ruby 2.7, at the same time, I started using safe operators (aka `&.`), thus breaking old versions of Ruby such as yours.

I do not recommend using pre 2.4 as it is not security supported (for instance your versions is not supported since Feb. 24th, 2016 : https://www.ruby-lang.org/en/news/2015/12/16/ruby-2-0-0-p648-released/ )

I did the fix however, and Release 2.3.1 might work for you (just published a few seconds ago), feel free to re-open if it still does not work.

**Comment by guilvey** [comment]
Hello @pierresouchay 
The version 2.3.1 you just delivered is working perfectly. Thanks a lot for your change and your reactivity. I can only agree with you about your remark on the ruby version I am using :-)
Guillaume.
